### PR TITLE
feat(677): implement promote command

### PR DESCRIPTION
**Pull request by tk3fftk**
This PR adds `sd-cmd promote namespace/name version tag` command.
It works locally like below:
- when we tag first time
image
- when we use already used tag
image
- (added) when we tag with same version
image

Related:
- design: https://github.com/screwdriver-cd/screwdriver/blob/master/design/commands.md 
- Issue: https://github.com/screwdriver-cd/screwdriver/issues/677

### CODE REVIEW

File: promoter/promoter.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+func (p *Promoter) Run() (err error) {
+	spec, err := p.sdAPI.GetCommand(p.smallSpec)
+	if err != nil {
+		fmt.Printf("%v does not exist yet.\n", p.tag)
+	} else {
+		fmt.Printf("Removing %v from %v\n", spec.Version, p.tag)
+	}
+	res, err := p.sdAPI.TagCommand(p.smallSpec, p.targetVersion, p.tag)
+	if err != nil {
+		return
```
Comment (yoshwata) [inline promoter/promoter.go]: It might better to show the tagged version is still that of before promoting if `TagCommand` failed.

Comment (tk3fftk) [reply promoter/promoter.go]: Added a message.

File: util/util.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ var caretRangesAndPinningRegexp = regexp.MustCompile(`^(\^)?\d(\.\d){2}$`)
 
 // tagRegexp check VERSION of Tags. Tags can only be named with A-Z,a-z,0-9,-
 // ex(latest stable feature-abc)
-var tagRegexp = regexp.MustCompile(`^[a-z][a-z0-9-]+$`)
+var tagRegexp = regexp.MustCompile(`^[a-zA-Z][\w-]+$`)
```
Comment (tk3fftk) [inline util/util.go]: This fix is based on https://github.com/screwdriver-cd/data-schema/blob/7554e482917ecb1964bcfcb4c6b656a2f6765c0f/config/regex.js

File: screwdriver/api/api.go [hunk 3]
```diff
@@ -272,6 +284,33 @@ func (c client) ValidateCommand(yamlString string) (*util.ValidateResponse, erro
...
+
+	uri, err := url.Parse(c.baseURL)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse URL: %v", err)
+	}
+	uri.Path = path.Join(uri.Path, "commands", commandSpec.Namespace, commandSpec.Name, "tags", tag)
+
+	responseBytes, statusCode, err := c.sendHTTPRequest("PUT", uri.String(), defaultContentType, body)
+	if err != nil {
+		return
```
Comment (s-yoshika) [inline screwdriver/api/api.go]: Is there no need to display any error here(and L305)?

Comment (s-yoshika) [reply screwdriver/api/api.go]: Thanks, I see.

**Pull request merged**